framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1210,71 +1223,137 @@ int pcm_drain(struct pcm *pcm)
 	}
 }
 
-static inline int pcm_mmap_avail(struct pcm *pcm)
-{
-	if (pcm == NULL) {
-		return 0;
-	}
-	return pcm_bytes_to_frames(pcm, pcm->next_size);
-}
-
+/** Application request to access a portion of direct (mmap) area
+ * @param[in] pcm A PCM handle
+ * @param[out] areas Returned mmap channel areas
+ * @param[out] offset Returned mmap area offset in frames
+ * @param[out] frames Mmap area portion size in frames
+ * @returns On success, zero is returned. On failure, a negative number returned
+ * @ingroup libtinyalsa-pcm
+ */
```

Comment:
already defined in header file, you can remove it.

Reply:
Ok